lib/taskRunner.ts

```diff
@@ -65,7 +64,12 @@ export class TaskRunner extends EventEmitter {
     config.specs = this.task.specs;
 
     if (this.runInFork) {
-      let deferred = q.defer();
+      let deferredResolve: (x?: any) => void;
```

Comment:
Similar to comments below, how about just wrapping the block in a promise and returning it